dashboard.py

```diff
@@ -1,41 +1,183 @@
-    bins = "-".join(bins)
-    print(bins)
-    print(n_velocity_bins)
-    dataset = TokenizedMidiDataset(
-        split="validation",
-        n_dstart_bins=n_dstart_bins,
-        n_duration_bins=n_duration_bins,
-        n_velocity_bins=n_velocity_bins,
+@hydra.main(version_base=None, config_path="config", config_name="dashboard_conf")
+def main(cfg: DictConfig):
+    with st.sidebar:
+        mode = st.selectbox(label="Display", options=["Model predictions", "Predict piece", "Tokenization review"])
+    if mode == "Tokenization review":
+        tokenization_review_dashboard()
+    if mode == "Predict piece":
+        predict_piece_dashboard(cfg)
+    if mode == "Model predictions":
+        model_predictions_review(cfg)
+
+
+def get_sample_info(dataset: BinsToVelocityDataset, midi_filename: str):
+    sample_data = dataset.dataset.filter(lambda row: row["midi_filename"] == midi_filename)
+    title, composer = sample_data["title"][0], sample_data["composer"][0]
+    return title, composer
+
+
+def model_predictions_review(cfg: DictConfig):
+    with st.sidebar:
+        # options
+        path = st.selectbox(label="model", options=glob.glob("models/*.pt"))
+        start_index = eval(st.text_input(label="start index", value="0"))
+
+    # load checkpoint
+    checkpoint = torch.load(path, map_location=cfg.device)
+    params = pd.DataFrame(checkpoint["cfg"]["model"], index=[0])
+    train_cfg = OmegaConf.create(checkpoint["cfg"])
+    st.markdown("Model parameters:")
+    st.table(params)
+
+    cols = st.columns(4)
+
+    with cols[0]:
+        st.markdown("### Unchanged")
+    with cols[1]:
+        st.markdown("### Quantized")
+    with cols[2]:
+        st.markdown("### Q. velocity")
+    with cols[3]:
+        st.markdown("### Predicted")
+
+    if cfg.dataset.dataset_name is None:
+        dataset = load_cached_dataset(train_cfg.dataset)
+    else:
+        cfg.dataset.bins = train_cfg.dataset.bins
+        cfg.dataset.sequence_size = train_cfg.dataset.sequence_size
+        dataset = load_cached_dataset(cfg.dataset, split=cfg.dataset_split)
+
+    model = make_model(
+        input_size=len(dataset.src_vocab),
+        output_size=len(dataset.tgt_vocab),
+        n=train_cfg.model.n,
+        d_model=train_cfg.model.d_model,
+        d_ff=train_cfg.model.d_ff,
+        h=train_cfg.model.h,
+        dropout=train_cfg.model.dropout,
     )
+    model.load_state_dict(checkpoint["model_state_dict"])
+
+    n_samples = 5
+    # predict velocities and get src, tgt and model output
+    print("Making predictions ...")
+    results = make_examples(dataset=dataset, model=model, start_index=start_index, n_examples=n_samples)
+
+    bins = train_cfg.dataset.bins.replace(" ", "-")
+    for it in range(n_samples):
+        # use every second record, so as not to create overlapped examples - it works together with make_examples()
+        idx = it * 2
+
+        src = results[it]["src"]
+        out = results[it]["out"]
```

Comment:
I think this `make_examples` usage is a very awkward pattern, that you're forced into because of the `Batch` class existence :)

The way I think this could work:

```python
n_samples = 10
idxs = np.random.randint(len(dataset), size=n_samples)
records = dataset[idxs]

for record in records:
    model_output = process_record(record)
    piece_info = json.loads(record["source"])
    
    piece_gt = MidiPiece(record)

    piece_pred = MidiPiece(record)
    piece_pred.df.velocity = model_output.velocity
    ...
```